iree/compiler/Dialect/VM/IR/test/CMakeLists.txt

```diff
@@ -12,4 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-iree_glob_lit_tests()
+# TODO(gcmn): should we generate the glob at bazel->cmake time instead?
+file(GLOB _TEST_FILES CONFIGURE_DEPENDS *.mlir)
```

Comment:
The documentation for [cmake glob](https://cmake.org/cmake/help/v3.12/command/file.html#filesystem) recommends against using it to gather sources.

I used CONFIGURE_DEPENDS, which should help, but it still suggests not doing it.

It's especially annoying to use because there doesn't seem to be a way to use it inline. We could instead have bazel_to_cmake evaluate the source tree. That would mean that you'd have to rerun bazel_to_cmake to get the updated cmake file. Unsure if that's better or worse.

Reply:
In https://github.com/google/iree/pull/562 I went with evaluating the glob at conversion time. @ScottTodd thoughts?

Reply:
My general preference with build systems is to be explicit about which source files are used, rather than rely on glob() to grab all files in a directory. Explicit control lets you have multiple libraries in a single package/folder so you can precisely tune your dependencies.

I'm less opinionated when it comes to these types of test files where we generally want to build/run all files in the directory without the risk of missing one in the build definition. If you miss a source file, the project doesn't build (unless you use dependency injection...); if you miss a test file, you have less coverage than you thought you had.

As for CMake globs, thanks for the link to the documentation. The configure step vs build step complicates things and I'm not sure how reliable `CONFIGURE_DEPENDS` is for the generators we may care about. If Bazel is primary build system and the Bazel -> CMake script gets used regularly, then evaluating the glob at Bazel -> CMake conversion time seems like a good solution to me.

Reply:
Yeah I also think glob is bad in most cases, but I think it makes sense for grabbing tests for exactly the reason you say. I'll stick with evaluating the glob on generation, which is what I implemented in #562 anyway :-)